src/specialty.jl

```diff
@@ -17,26 +17,485 @@ end
+[`NoPad`](@ref)
+
+### `"reflect"`
+The border pixels reflect relative to the edge itself.
+```math
+\\boxed{
+\\begin{array}{l|c|r}
+  d\\, c\\, b\\, a  &  a \\, b \\, c \\, d \\, e \\, f & f \\, e \\, d \\, c
+\\end{array}
+}
+```
+See also: [`Pad`](@ref), [`padarray`](@ref), [`Inner`](@ref) and
+[`NoPad`](@ref)
+
+# Example
+
+This example compares the quality of the gradient estimation methods in terms of
+the accuracy with which the orientation of the gradient is estimated.
+
+```julia
+using Images
+
+values = linspace(-1,1,128);
+w = 1.6*pi;
+
+# Define a function of a sinusoidal grating, f(x,y) = sin( (w*x)^2 + (w*y)^2 ),
+# together with its exact partial derivates.
+I = [sin( (w*x)^2 + (w*y)^2 ) for y in values, x in values];
+Ix = [2*w*x*cos( (w*x)^2 + (w*y)^2 ) for y in values, x in values];
+Iy = [2*w*y*cos( (w*x)^2 + (w*y)^2 ) for y in values, x in values];
+
+# Determine the exact orientation of the gradients.
+direction_true = atan.(Iy./Ix);
+
+for kernelfunc in (KernelFactors.prewitt, KernelFactors.sobel,
+                   KernelFactors.ando3, KernelFactors.scharr,
+                   KernelFactors.bickley)
+
+# Estimate the gradients and their orientations.
+Gy, Gx = imgradients(I,kernelfunc, "replicate");
+direction_estimated = atan.(Gy./Gx);
+
+# Determine the mean absolute deviation between the estimated and true
+# orientation. Ignore the values at the border since we expect them to be
+# erroneous.
+error = mean(abs.(direction_true[2:end-1,2:end-1] -
+                 direction_estimated[2:end-1,2:end-1]));
+
+error = round(error,5);
+println("Using \$kernelfunc results in a mean absolute deviation of \$error")
+end
+
+# output
+
+Using ImageFiltering.KernelFactors.prewitt results in a mean absolute deviation of 0.01069
+Using ImageFiltering.KernelFactors.sobel results in a mean absolute deviation of 0.00522
+Using ImageFiltering.KernelFactors.ando3 results in a mean absolute deviation of 0.00365
+Using ImageFiltering.KernelFactors.scharr results in a mean absolute deviation of 0.00126
+Using ImageFiltering.KernelFactors.bickley results in a mean absolute deviation of 0.00038
+```
+# References
+  1. Bernd, J. 2012. *Digital Image Processing* (7th ed.). Springer Publishing Company, Incorporated
+  2. Patra, M. and Karttunen, M. (2006), *Stencils with isotropic discretization error for differential operators*. Numer. Methods Partial Differential Eq., 22:936–953. doi:10.1002/num.20129
+  3. Belyaev, A. (2013). *Implicit Image Differentiation and Filtering with Applications to Image Sharpening*. SIAM Journal on Imaging Sciences, 6(1), 660–679. doi:10.1137/12087092x
+  4. Bickley, W. G. (1948), *Finite difference Formulae for the square lattice*. The Quarterly Journal of Mechanics and Applied Mathematics, 1(1), 35–42. doi:10.1093/qjmam/1.1.35
+  5. Scharr, H., and Weickert, J. (2000). *An Anisotropic Diffusion Algorithm with Optimized Rotation Invariance*. Mustererkennung 2000, 460–467. doi:10.1007/978-3-642-59802-9_58
+---
```

Comment:
The full reference for the Ando kernels is

```
Ando, S. (2000), *Consistent gradient operators*, IEEE Transactions on Pattern Analysis and Machine Intelligence, 22(3), 252-265. doi: 10.1109/34.841757
```

Can you add it here and add a citation above?